viewer/packages/tools/src/Timeline/TimelineTool.test.ts

```diff
@@ -0,0 +1,168 @@
+    kf3Appearance = { renderGhosted: false };
+
+    const kf1 = timelineTool.createKeyframe(new Date('2021-10-25'));
+    const kf2 = timelineTool.createKeyframe(new Date('2021-10-26'));
+    const kf3 = timelineTool.createKeyframe(new Date('2021-10-27'));
+
+    kf1.assignStyledNodeCollection(kf1Collection, kf1Appearance);
+    kf2.assignStyledNodeCollection(kf2Collection, kf2Appearance);
+    kf3.assignStyledNodeCollection(kf3Collection, kf3Appearance);
+  });
+
+  test('play() playback the Timeline', () => {
+    const assignStyledNodeCollectionSpy = jest.spyOn(model, 'assignStyledNodeCollection');
+    const unassignStyledNodeCollectionSpy = jest.spyOn(model, 'unassignStyledNodeCollection');
+
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 30000);
+    const current = TWEEN.now();
+    TWEEN.update(current + 10000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf1Collection, kf1Appearance);
+    TWEEN.update(current + 20000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf2Collection, kf2Appearance);
+    TWEEN.update(current + 30000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf3Collection, kf3Appearance);
+    TWEEN.update();
+
+    expect(unassignStyledNodeCollectionSpy).toBeCalledWith(kf2Collection);
+  });
+
+  test('stop() stops the Timeline playback', () => {
+    const assignStyledNodeCollectionSpy = jest.spyOn(model, 'assignStyledNodeCollection');
+
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 40000);
+    const current = TWEEN.now();
+    TWEEN.update(current + 10000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf1Collection, kf1Appearance);
+    TWEEN.update(current + 20000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf2Collection, kf2Appearance);
+    TWEEN.update(current + 30000);
+
+    timelineTool.stop();
+
+    expect(TWEEN.update()).toBeFalse();
+  });
+
+  test('pause() & resume() pauses & resumes the Timeline', () => {
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 40000);
+    const current = TWEEN.now();
+    TWEEN.update(current + 10000);
+
+    expect(TWEEN.update()).toBeTrue();
+
+    timelineTool.pause();
+    expect(TWEEN.update()).toBeFalse();
+    TWEEN.update(current + 20000);
+
+    timelineTool.resume();
+    expect(TWEEN.update()).toBeTrue();
+    TWEEN.update(current + 30000);
+  });
+
+  test('play() while play is active', () => {
+    const stopSpy = jest.spyOn(timelineTool, 'stop');
+    const assignStyledNodeCollectionSpy = jest.spyOn(model, 'assignStyledNodeCollection');
+    const unassignStyledNodeCollectionSpy = jest.spyOn(model, 'unassignStyledNodeCollection');
+
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 30000);
+    const current = TWEEN.now();
+    TWEEN.update(current + 10000);
+
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 30000);
+
+    expect(stopSpy).toBeCalledTimes(2);
```

Comment:
I think this is testing a bit too much internal stuff. I'd rather just test that only one call to `assignStyledNodeCollection` is triggered when calling `TWEEN.update()`. Also, the expect you have below could then be removed - it's best to test few expectations per test to make the test easier to read.